src/dart/codegen.rs

```diff
@@ -11,80 +11,102 @@ impl Codegen {
     }
 
     pub fn codegen_items(&mut self, items: &[Item]) -> Vec<Node<ast::Item>> {
-        items.iter().map(|item| self.codegen_item(item)).collect()
+        items
+            .iter()
+            .flat_map(|item| self.codegen_item(item))
+            .collect()
     }
 
-    fn codegen_item(&mut self, item: &Item) -> Node<ast::Item> {
+    fn codegen_item(&mut self, item: &Item) -> Vec<Node<ast::Item>> {
         match *item {
             Item::ComponentDef {
-                name,
+                mut name,
                 ref fields,
                 ref dart_members,
                 ref body,
             } => {
-                let params;
-                let superclass;
+                let mut items = vec![];
+                let mut superclass = Some(ast::Type::simple_path("StatelessWidget"));
                 let mut class_members = vec![];
-                superclass = Some(ast::Type::simple_path("StatelessWidget"));
-                if !fields.is_empty() {
-                    let mut args = vec![ast::ArgDef::simple(ast::Type::simple_path("Key"), "key")];
-
-                    for field in fields {
-                        let mut var_ty = Node::new(ast::Type::Infer);
-                        if let Some(ref ty) = field.ty {
-                            var_ty = self.codegen_type(ty);
-                        }
-                        args.push(ast::ArgDef {
-                            metadata: vec![],
-                            covariant: false,
-                            ty: ast::VarType {
-                                fcv: ast::FinalConstVar::Var,
-                                ty: var_ty,
-                            },
-                            field: true,
-                            var: Node::new(ast::VarDef {
-                                name: field.name,
-                                init: field
-                                    .default
-                                    .as_ref()
-                                    .map(|default| self.codegen_expr(default)),
-                            }),
-                        });
-                    }
-                    params = ast::FnSig {
-                        return_type: Node::new(ast::Type::Infer),
-                        required: vec![],
-                        optional: args,
-                        optional_kind: ast::OptionalArgKind::Named,
-                        async: false,
-                        generator: false,
+                if fields.iter().any(|x| x.mutable) {
```

Comment:
Whenever you have something like this, `f` or `field` instead of `x` would be helpful.